### DISCUSSION TITLE
Is self-hosting an option?

### DISCUSSION THREAD

**Discussion by alexgleason** [skymethod/denoflare#44] [Q&A] [answered]
I want to develop a Deno application that can be deployed to Cloudflare Workers OR self-hosted, depending on the user's preference. Is that easy to do with denoflare?

**Comment by johnspurlock-skymethod** [comment, marked as answer]
Depends on how many Cloudflare-specific services the worker uses (like KV, R2, Cache API etc) which need to have local implementations.

[`denoflare serve`](https://denoflare.dev/cli/serve) essentially runs the worker in a local Deno process, with the ability to provide mock or local implementations for various CF services, it may work well for what you need, or you could create a similar static server loop [based on it](https://github.com/skymethod/denoflare/blob/master/cli/cli_serve.ts).

**Reply by alexgleason** [reply]
Thank you!
